src/main/java/frc/robot/commands/AlignCommand.java

```diff
@@ -31,21 +56,28 @@ public AlignCommand(Limelight limelight, DriveSubsystem driveSubsystem) {
                         EntryListenerFlags.kUpdate & EntryListenerFlags.kNew & EntryListenerFlags.kImmediate);
 
         // Degrees, degrees / second
-        aimyAim.setTolerance(3, 0.5);
-        aimyAim.enableContinuousInput(-180, 180);
-
-        addRequirements(driveSubsystem);
+        pid.setTolerance(3, 0.5);
+        pid.enableContinuousInput(-180, 180);
     }
 
     @Override
     public void execute() {
-        double output = aimyAim.calculate(limelight.getYaw(), 0);
+        // turning left is negative, right is positive
+        double offsetAngle;
+        if (useLimelight) offsetAngle = limelight.getYaw();
+        else {
+            // use gyroscope
+            double currentAngle = navx.getAngle();
+            offsetAngle = turnToAngle - currentAngle;
+        }
+
+        double output = pid.calculate(offsetAngle, 0);
```

Comment:
This seems wacky. You always use 0 as the setpoint, but you also provide this command with a turnToAngle? Why not use the turnToAngle as the setpoint?

Reply:
I was just following what was already in the alignCommand class when only the Limelight was being used, but after looking at the PIDController documentation I'm pretty sure doing pid.calculate(offsetAngle, 0) is the same thing as pid.calculate(offsetAngle). Once I break this into two separate commands I can just use pid.calculate(limelight.getYaw()) and pid.calculate(navx.getAngle(), turnToAngle).